packages/gestalt/src/SheetMobile/PartialPage.js

```diff
@@ -0,0 +1,325 @@
+    if (window && window.body?.style?.overflow) {
+      window.body.style.overflow = 'hidden';
+    }
+    return () => {
+      if (window && window.body?.style?.overflow) {
+        window.body.style.overflow = 'auto';
+      }
+    };
+  }, []);
+
+  useEffect(() => {
+    window.addEventListener('resize', updateShadows);
+    return () => {
+      window.removeEventListener('resize', updateShadows);
+    };
+  }, [updateShadows]);
+
+  // Use useLayoutEffect instead of useEffect as we need to close the component synchronously after all DOM mutations, useEffect was needed to prevent changing state while still rendering but useEffect will create a ms blink of the full OverlayPanel after closing which gets prevented with useLayoutEffect
+  useLayoutEffect(() => {
+    if (animationState === ANIMATION_STATE.unmount) {
+      onDismiss();
+    }
+  }, [animationState, onDismiss]);
+
+  return (
+    <StopScrollBehavior>
+      <TrapFocusBehavior>
+        <div
+          id={id}
+          aria-label={accessibilityLabel ?? defaultAccessibilityLabel}
+          className={classnames(
+            sheetMobileStyles.container,
+            sheetMobileStyles.partialPageContainer,
+          )}
+          role={role}
+        >
+          <Backdrop closeOnOutsideClick={closeOnOutsideClick} onClick={handleBackdropClick}>
+            <div
+              className={classnames(sheetMobileStyles.wrapper, focusStyles.hideOutline, {
+                [sheetMobileStyles.defaultWrapper]: size === 'default',
+                [sheetMobileStyles.autoWrapper]: size === 'auto',
+                [animation.animationInBottom]: animationState === ANIMATION_STATE.animatedOpening,
+                [animation.animationOutBottom]: animationState === ANIMATION_STATE.animatedClosing,
+              })}
+              onAnimationEnd={handleOnAnimationEnd}
+              tabIndex={-1}
+              style={{ width: '100%' }}
+            >
+              <Box position="relative" display="flex" direction="column" width="100%">
+                <Box
+                  padding={4}
+                  borderStyle={showTopShadow ? 'raisedTopShadow' : undefined}
+                  position="relative"
+                  fit
+                >
+                  <Flex justifyContent="center">
+                    <Box marginBottom={2}>
+                      <TapArea
+                        fullWidth={false}
+                        ref={grabberRef}
+                        rounding={7}
+                        accessibilityLabel={accessibilityGrabberLabel}
+                      >
+                        <Box height={5} width={37} color="secondary" rounding={7} />
+                      </TapArea>
+                    </Box>
+                  </Flex>
+                  <Flex justifyContent="center" alignItems="center" gap={4}>
```

Comment:
Since all of these looks the same w/ FullPage and PartialPage, I think we should just put them as part of the same component, so we don't have to change things in two places. Maybe `SheetMobileContent`

Reply:
Actually, maybe two components. `SheetMobileContent` and `SheetMobileHeader`, it may be quite possible we can reuse the header in other places (like Modal)

Reply:
Ok. I was planning on drying more later... cos there's A LOT to dry here. But I'll look into this.

Reply:
DONE. I can dry more, but leaving as is for now